homeassistant/components/homeassistant/__init__.py

```diff
@@ -251,15 +252,23 @@ async def async_handle_reload_config(call: ha.ServiceCall) -> None:
     async def async_set_location(call: ha.ServiceCall) -> None:
         """Service handler to set location."""
         await hass.config.async_update(
-            latitude=call.data[ATTR_LATITUDE], longitude=call.data[ATTR_LONGITUDE]
+            latitude=call.data[ATTR_LATITUDE],
+            longitude=call.data[ATTR_LONGITUDE],
+            elevation=call.data.get(ATTR_ELEVATION, None),
```

Comment:
This would remove any current elevation.
Should be changed so parameter is only set if service call data includes elevation